hs_composite_resource/models.py

```diff
@@ -0,0 +1,125 @@
+import os
+
+from django.core.exceptions import ValidationError
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def get_metadata_xml(self, pretty_print=True):
+        from lxml import etree
+
+        # get resource level core metadata as xml string
+        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=False)
+        # add file type metadata xml
+
+        # create an etree xml object
+        RDF_ROOT = etree.fromstring(xml_string)
+
+        # get root 'Description' element that contains all other elements
+        container = RDF_ROOT.find('rdf:Description', namespaces=self.metadata.NAMESPACES)
+
+        for lf in self.logical_files:
+            lf.metadata.add_to_xml_container(container)
+
+        return etree.tostring(RDF_ROOT, pretty_print=pretty_print)
+
+    def check_folder_creation(self, folder_full_path):
```

Comment:
This code should be generic and available to all resources. See #1495 for implementation of supports_folders, which would switch this on and off. Eventually this needs to be upstreamed.